datastash/README.md

```diff
@@ -0,0 +1,76 @@
+
+```
+curl -XPUT -H 'Content-Type: application/json' "<target>:<port>/_index_template/datastash_template_<index-name>?pretty" -d @/tmp/datastash_template.json
+```
+
+6. Finally, clean up the temporary files we created:
+
+```
+rm /tmp/settings.json /tmp/mappings.json /tmp/datastash_template.json
+```
+
+### 3. Run the migration
+
+You're finally ready to kick off the migration! Move your data by running `docker compose up`. Once the process completes, the docker container will shut itself down.
+
+# Limitations
+
+This tool has several limitations:
+
+1. Logstash does not include any support for clustering or load balancing ([[1]](https://discuss.elastic.co/t/how-can-make-a-cluster-with-logstash-is-it-possible/250444/2), [[2]](https://github.com/elastic/logstash/issues/2632), [[3]](https://github.com/elastic/logstash/issues/2633)), so users must set up their own load balancing. This makes it impossible to horizontally scale a Logstash setup that uses a pull-based plugin like the ElasticSearch/OpenSearch input plugin.
+
+1. Since Logstash cannot make any assumptions about the input or output, it cannot perform any setup steps before sending data to the target cluster, and the validation it performs is minimal.
+    * Logstash does not validate the configuration of the output plugin. Any mistakes in the OpenSearch output plugin configuration (endpoint, username/password, SSL, etc.) do not fail the Logstash pipeline immediately. Instead, Logstash will continue to process the input (until complete), and then fail each batch at the output.
+    * Contrary to the output plugin, Logstash does validate connectivity from the input plugin. Any errors therein cause the pipeline to stop processing completely. However, since Logstash cannot make any assumptions about the input, it does not check for a index of the same name on the target cluster.
+    * Further, these decisions mean that index mappings are not copied to the target cluster, which necessitates the manual steps outlined above.
+
+1. Logstash also implicitly adds two internal fields to each record in its output - “@version” and “@timestamp”. There is no way to prevent Logstash from adding these fields. The suggested workaround is to apply a mutate filter to remove these fields, but such an approach would remove these fields even if they existed in the original index.
+
+1. Logstash versions above 7.14 do not support OSS endpoints due to a [change introduced to the ElasticSearch clients](https://discuss.elastic.co/t/logstash-not-start-after-upgrade-to-new-version/291849/2). In order to work around this restriction, this tool uses an older image ([7.13.4](https://hub.docker.com/layers/opensearchproject/logstash-oss-with-opensearch-output-plugin/7.13.4/images/sha256-a0370926a62f5a81078e24f9acb37e6750ee2bc8472a37b071c40b6765319ea5?context=explore)) from Dockerhub. This prevents the client from receiving any further updates, which risks breaking compatibility with future versions of ElasticSearch.
```

Comment:
Markdown can usually handle this intelligently - you'll see that the [rendered file](https://github.com/kartg/opensearch-migrations/tree/datastash/datastash#limitations) has the correct numbering.

The reason the previous sections need to use explicit numbering is because either:

1. They're [sub-headings](https://github.com/kartg/opensearch-migrations/tree/datastash/datastash#0-prerequisites), so the increment logic doesn't trigger
1. The [items](https://github.com/kartg/opensearch-migrations/tree/datastash/datastash#2-configure-index-templates-on-the-target-cluster) are separated by `multi-line code blocks`, so the increment logic doesn't work correctly

(Easter egg - the source of the numbered list above also uses 1 and 1 😉 )